### DISCUSSION TITLE
run wsl from  asp.net

### DISCUSSION THREAD

**Discussion by CheckingCode** [microsoft/WSL#6320] [General]
**WSL logs**: 

#  Expected behavior

# Actual behavior

**Comment by CheckingCode** [comment]
<!--
Microsoft Windows [Version 10.0.19042.630]
Linux version 4.19.128-microsoft-standard (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Tue Jun 23 12:58:10 UTC 2020
Create Asp.net form project
add  <asp:Button ID="Button" runat="server"
add protected void Button_Click(object sender, EventArgs e){   
 //-----------------------------------------------version2
            Runspace runspace = RunspaceFactory.CreateRunspace();
            runspace.Open();
            Pipeline pipeline = runspace.CreatePipeline();
            //pipeline.Commands.AddScript(@"bash -c ""C:\Users\Admin\test_new.sh""");
           
            pipeline.Commands.AddScript(@" C:\Code\test_new\grade.bat");

     }
inside of grade.bat i try call WSL
wsl.exe ./grade.sh 

get error : linux - wsl' is not recognized as an internal or external ...
ok i try run cmd or powershell from windows and run C:\Code\test_new\grade.bat command
all ok its work wsl 👍 
if run this code from asp.net  pipeline.Commands.AddScript(@" cmd.exe");
i get command windows CMD 
run: cd C:\Windows\System32\
run in this cmd windows  : dir 
i see directory but i dont have WSL.exe file in the directory
go to windows open explorer C:\Windows\System32\
i see  WSL.exe   ???????????????????????
ok inside of grade.bat i try call WSL
C:\Windows\Sysnative\wsl.exe 
all ok i run wsl from asp.net
but its only iis expess if try publish project to IIS and run from web WSL on localhost 
get error : linux - wsl' is not recognized as an internal or external ...

**Comment by sirredbeard** [comment]
I am not sure what you are asking. You can run ASP.NET on WSL, we did it in our demo on WSL 1 for Build 2019.

**Reply by CheckingCode** [reply]
run WSL on ASP.net server side 
Computer Windows 
IIS server 
ASP.net or Core web project
From web page ASP run on server side: call WSL/SWL2  run linux programm , return result to the web page
Problem after transfer project from IIS express to IIS 
WSL.exe have not user iis app pool

**Comment by therealkenc** [comment]
Taking ["wsl from"](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.start?view=net-5.0) at face value, and playing guess the fail, maybe variation #4652.
